### PR TITLE
[EGM HLT@Phase2] Move to TICL v4

### PR DESCRIPTION
**Pull request by swagata87**
#### PR description:

TICL V4 was introduced by HGCAL DPG by this PR: https://github.com/cms-sw/cmssw/pull/38880
However, Phase2 HLT still uses TICL v3.
This PR aims to make the TICL v3->v4 transition for e/gamma HLT in Phase2.
Note that other parts of Phase2 simplified HLT menu (like Jet paths etc) still use TICL v3, and those are not touched in this PR.
  
#### PR validation:
`runTheMatrix.py -l 39434.75` ran fine.

Measured efficiency of HLT_Ele32 w.r.t gen electrons using this sample: `/ZprimeToEE_M-6000_TuneCP5_14TeV-pythia8/PhaseIISpring22DRMiniAOD-PU200_123X_mcRun4_realistic_v11-v1/GEN-SIM-DIGI-RAW-MINIAOD`. Made a comparison between TICL v3 and TICL v4. There is some improvement in v4, as seen in the plot, in HGCAL region.

Using the same sample, some showershape variables at HLT were checked. The small changes in shape look okay.

### CODE REVIEW

File: HLTrigger/Configuration/python/HLT_75e33/tasks/HLTHgcalTiclPFClusteringForEgammaUnseededTask_cfi.py [hunk 1]
```diff
@@ -13,14 +12,12 @@
 from ..modules.particleFlowSuperClusterHGCalFromTICLUnseeded_cfi import *
 from ..modules.ticlLayerTileProducer_cfi import *
 from ..modules.ticlSeedingGlobal_cfi import *
-from ..modules.ticlTrackstersEMForEgamma_cfi import *
-from ..modules.ticlTrackstersHADForEgamma_cfi import *
```
Comment (missirol) [inline HLTrigger/Configuration/python/HLT_75e33/tasks/HLTHgcalTiclPFClusteringForEgammaUnseededTask_cfi.py:17]: Can files like `ticlTrackstersEMForEgamma_cfi` be removed then?

Comment (swagata87) [reply HLTrigger/Configuration/python/HLT_75e33/tasks/HLTHgcalTiclPFClusteringForEgammaUnseededTask_cfi.py:17]: indeed, they are not being used anywhere.. so can be removed.
I'll clean it up then

**Comment by missirol** [comment]
test parameters:
 - workflows = 20834.76
 - relvals_opt = --what cleanedupgrade,standard,highstats,pileup,generator,extendedgen,production,ged,machine,premix,nano

**Comment by missirol** [comment]
please test

**Comment by missirol** [comment]
+hlt

 - small differences in E/gamma triggers for wfs using the HLT Phase-2 menu (see [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_12_6_X_2022-10-22-1100+a02e97/53602/triggerResults/)), which is compatible with the physics changes described in this PR

**Comment by perrotta** [comment]
+1

**Pull request merged**